src/main/java/org/mozilla/jss/JSSProvider.java

```diff
@@ -32,10 +34,9 @@
                                             JSS_PATCH_VERSION)/10000.0;
 
     private static final List<String> DEPRECATED_ALGORITHMS = Arrays.asList(
-            "SHA-1",
-            "SHA1",
-            "SHA_1",
-            "SHA");
+            "SHA\\s*$|SHA\\s*,|SHAwith|SHA1|SHA-1|SHA_1", // SHA-1 and variations thereof
```

Comment:
The `SHA\\s*,` is needed to check for SHA in config files since it might be part of a list, but it's probably not needed here since IIUC `MessageDigest.getInstance()` takes only a single algorithm.

The `SHA\\s*$` probably could be simplified to `SHA$` since people probably wouldn't have spaces in the algorithm name, or normalize it first.

Reply:
Yeah good points.